### ISSUE TITLE
Update clang-format to properly break braces before `typedef enum {`

### ISSUE THREAD

**Issue by updateing** [microsoft/vscode-cpptools#1020]
This is an [upstream bug](https://reviews.llvm.org/D37143?id=113035) which was fixed two weeks earlier. Currently I'm able to reproduce this on

* Ubuntu 16.04
* VSCode 1.15.1 stable build
* Cpptools 0.12.3
* `BreakBeforeBraces: Allman` in .clang-format (also reproducible with default "Visual Studio" style)

with following code snippet:

```c
typedef enum _my_enum {
    ENUM_VALUE_MIN,
    ENUM_VALUE_MAX,
} my_enum_t;
```

VSCode would format it as I've typed above. The expected result is

```c
typedef enum _my_enum
{
    ENUM_VALUE_MIN,
    ENUM_VALUE_MAX,
} my_enum_t;
```

Could you please update clang-format to include fix for this bug? Thanks!

**Comment by sean-mcmanus** [comment]
As a workaround, you can set your C_Cpp.clang_format_path to the path to a local version of clang-format that has the fix. Do you know which version of clang-format has this fix in it? We're currently shipping 4.0, but I see that 5.0 is now available.

**Comment by updateing** [comment]
I just checked the 5.0 release, only to find that the fix was not included. Guess I'll have to build one myself. Thanks!
